File: 2022/0627/PS_구명보트_이길상.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        // 투 포인터
        int low = 0, high = people.length - 1;
        
        // 정렬
        Arrays.sort(people);
        
        // 보트 시뮬레이션
        while(low < high){
            // 2명
            if(people[low] + people[high] <= limit){
                low++;
                high--;
            }
            // 1명
            else{
                high--;
            }
            answer++;
        }
        
        if(low == high)
            answer++;
        
        return answer;
    }
}

File: 2023/0214구명보트/PS_구명보트_이길상.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42885

def solution(people, limit):
    # 오름차순 정렬
    people.sort()
    
    # 투 포인터
    low = 0
    high = len(people)-1
    
    answer = 0
    while low < high:
        # 2명 탈수있는 경우
        if people[low] + people[high] <= limit:
            low+=1
            high-=1
        # 1명 탈수있는 경우
        else:
            high-=1
            
        answer += 1
    
    # 1명 남은 경우
    if low == high:
        answer += 1
    
    return answer
